nmap_dns.cc

```diff
@@ -820,9 +806,14 @@ static void read_evt_handler(nsock_pool nsp, nsock_event evt, void *) {
     }
   }
 
-  if (!processing_successful){
-    sockaddr_storage discard;
-    process_result(discard, "", ACTION_SYSTEM_RESOLVE, p.id);
+  if (!processing_successful) {
+    if (DNS_HAS_FLAG(f, DNS::TRUNCATED)) {
+      sockaddr_storage discard;
+      process_result(discard, "", ACTION_SYSTEM_RESOLVE, p.id);
+    }
+    else
```

Comment:
Just a personal preference, but please use braces `{}` to wrap this else clause. Chaining `else` to `if` to `log_write` leaves lots of places for someone to later add some code intended to fit in the `else` which actually runs unconditionally.